File: java/parser/src/main/java/gov/lbl/parser/parser/Extends_clauseVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.parser.domain.Annotation;
import gov.lbl.parser.domain.Class_modification;
import gov.lbl.parser.domain.Extends_clause;
import gov.lbl.parser.domain.Name;
import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;

public class Extends_clauseVisitor extends modelicaBaseVisitor<Extends_clause> {
    @Override
    public Extends_clause visitExtends_clause(modelicaParser.Extends_clauseContext ctx) {
        NameVisitor nameVisitor = new NameVisitor();
        Name name = ctx.name() == null ? null : ctx.name().accept(nameVisitor);

        Class_modificationVisitor class_modificationVisitor = new Class_modificationVisitor();
        Class_modification class_modification = ctx.class_modification() == null ? null : ctx.class_modification().accept(class_modificationVisitor);
        AnnotationVisitor annotationVisitor = new AnnotationVisitor();
        Annotation annotation = ctx.annotation() == null ? null : ctx.annotation().accept(annotationVisitor);      
        
        return new Extends_clause(name, class_modification, annotation);
    }
}

File: jsParser/parser/Extends_clauseVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Extends_clause = require('../domain/Extends_clause')

const NameVisitor = require('./NameVisitor')
const Class_modificationVisitor = require('./Class_modificationVisitor')
const AnnotationVisitor = require('./AnnotationVisitor')

class Extends_clauseVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitExtends_clause (ctx) {
    let name = null
    let class_modification = null
    let annotation = null

    if (ctx.name()) {
      const nameVisitor = new NameVisitor.NameVisitor()
      name = nameVisitor.visitName(ctx.name())
    }

    if (ctx.class_modification()) {
      const class_modificationVisitor = new Class_modificationVisitor.Class_modificationVisitor()
      class_modification = class_modificationVisitor.visitClass_modification(ctx.class_modification())
    }

    if (ctx.annotation()) {
      const annotationVisitor = new AnnotationVisitor.AnnotationVisitor()
      annotation = annotationVisitor.visitAnnotation(ctx.annotation())
    }

    return new Extends_clause.Extends_clause(name, class_modification, annotation)
  }
};

Extends_clauseVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitExtends_clause = this.visitExtends_clause
exports.Extends_clauseVisitor = Extends_clauseVisitor
